### PR TITLE
[Embedding][Serve] Phase 3: first-class encoder embedding path

### PR DESCRIPTION
**Pull request by xthomaswang**
Part of #3451.

Builds on #3461.

Depends on apache/tvm#19392.

## Summary

Completes embedding phase 3 from #3451: encoder embedding models are now a first-class serve/runtime path in MLC LLM.

Phase 2 (#3461) established the metadata-driven single-task embedding runtime boundary on the Python side. This PR pushes the encoder embedding path further down into the C++ serving/runtime layer by adding a dedicated embedding request lane, encoder batch prefill action, and threaded-engine / FFI plumbing, while keeping the normal chat path structurally unchanged.

It also switches BERT encoder attention to a valid-lens-based path backed by TVM's masked sequence prefill helper.

This PR is focused on the encoder embedding path only. Decoder embedding is not implemented in this lane yet.

### C++ serving/runtime

- `cpp/metadata/model.cc`
  - allow embedding models to omit `kv_cache` metadata
- `cpp/serve/data.{h,cc}`
  - add `EmbeddingRequest` / `EmbeddingResult` / request-state types and FFI packing
- `cpp/serve/engine.{h,cc}`
  - add embedding request APIs
  - dispatch chat vs embedding lanes by `model_task`
- `cpp/serve/engine_actions/action.h`
- `cpp/serve/engine_actions/batch_embedding_prefill.cc`
  - add the encoder batch embedding action
- `cpp/serve/engine_state.{h,cc}`
  - add embedding queue / request state / callback storage
- `cpp/serve/function_table.{h,cc}`
  - load encoder `prefill` from the compiled model
- `cpp/serve/model.{h,cc}`
  - add encoder prefill + pooling helpers
- `cpp/serve/threaded_engine.{h,cc}`
  - expose the embedding request path through the threaded engine / FFI

### Python model/op/runtime

- `python/mlc_llm/model/bert/bert_model.py`
  - switch BERT self-attention to `encoder_attention(...)`
- `python/mlc_llm/op/__init__.py`
- `python/mlc_llm/op/attention.py`
  - add `encoder_attention(q, k, v, valid_lens)`
- `python/mlc_llm/serve/embedding_engine.py`
  - route embedding requests through the new threaded encoder runtime

### Tests

- `tests/python/serve/test_embedding_engine.py`
  - add coverage for the new encoder embedding path

## TVM Dependency

This PR depends on apache/tvm#19392.

The direct dependency is in:

- `python/mlc_llm/op/attention.py`
- `python/mlc_llm/model/bert/bert_model.py`
- `encoder_attention(...)` uses TVM's `_attention_sequence_prefill_with_mask`, which is introduced in apache/tvm#19392. Without that TVM change, encoder embedding would not be able to run through the new masked batch-prefill path and would instead fall back to sequential per-request prefill.

## Notes

- chat vs embedding dispatch is now driven by `model_task`
- this PR implements the encoder embedding lane only
- no `/v1/embeddings` API contract change is intended in this PR

### CODE REVIEW

File: cpp/serve/model.cc [hunk 1]
```diff
@@ -1185,6 +1188,188 @@ class ModelImpl : public ModelObj {
...
+        << "Encoder prefill function is not available in this model.";
+    // input_ids_nd: CPU [batch_size, max_len] int32
+    // attention_mask_nd: CPU [batch_size, max_len] int32
+    TVM_FFI_ICHECK_EQ(input_ids_nd->ndim, 2);
+    TVM_FFI_ICHECK_EQ(attention_mask_nd->ndim, 2);
+
+    // Copy 2D tensors to device. We allocate fresh contiguous device tensors
+    // to avoid non-contiguity issues from CopyToWorker0's view-based caching
+    // (which creates non-contiguous views when actual shape != reserved shape).
+    Tensor input_ids_device = Tensor::Empty(input_ids_nd.Shape(), input_ids_nd.DataType(), device_);
```
[the remark this answers is not in the document]
Comment (xthomaswang) [reply cpp/serve/model.cc:1216]: not a major performance problem in encoder path.
